### ISSUE TITLE
About the training configuration of cifar10

### ISSUE THREAD

**Issue by xinyuxiao** [haiduo/Double-Rounding#1]
dear sir, i train the model on f32 (resnet20) with 200 epoch，but the top-1 is only 89% (compared to 92.3%). Is there something I didn't notice? thank you very much!

**Comment by haiduo** [comment]
Hi @xinyuxiao ,

If you are using the same hyperparameter settings as in my paper, the performance shouldn't be this low. Perhaps you can check whether the random seed, learning rate, weight decay, and batch size are consistent? Try adjusting the learning rate or the random seed. You can refer to the ablation settings I provided below:

![Image](https://github.com/user-attachments/assets/bd6a0e8f-9029-4923-a551-14420c5743f4)

**Comment by xinyuxiao** [comment]
ok, thank you very much

**Comment by xinyuxiao** [comment]
@haiduo

**Comment by xinyuxiao** [comment]
Dear @haiduo 
if the config is set as below (depending on this config, the top-1  is lower to 64%), my main confusion is whether lr is 1e-1 (when lr=1e-3, the top-1 is up to 88.88%). If there are other setting problems, please point them out. Thank you very much.

PSKD=False, adam_beta1=0.9, adam_beta2=0.999, alpha_T=0.7, batch_size=128, bit_width_list='32', dataset='cifar10', double_optimizer=False, epochs=90, evaluate=False, fixed_lr=True, lr=0.1, lr_decay='20,40,60', lr_min=1e-10, lr_scheduler='CosineAnnealingLR', manu_gpu=0, max_norm=1.0, model='resnet20q', multi_lr=False, multi_lr_alpha=0.1, multi_lr_stop_bit=0, multiprocessing_distributed=False, node=1, optimizer='adam', pretrain=False, print_freq=100, rank=0, repeated_aug=False, results_dir='code/Multi_Precision/results', resume='', save_name='resnet20q_32bits', scale_no_wd=False, seed=42, sgd_dampening=0, sgd_momentum=0.9, start_epoch=0, statistic_gradient=True, step_inside=True, sync_bn=False, train_url='', type_a='lsq', type_w='lsq', warm_up=False, warmup_epoch=10, weight_decay=0.0005, workers=4, world_size=1

**Comment by haiduo** [comment]
Hi @xinyuxiao ,

Yes, the learning rate is lr=1e-1. If you still encounter issues, I’ll try to re-implement it later and open-source the checkpoints I used. Recently, I’m working on a conference submission, so the upload might be a bit delayed — probably after the 17th.

Please wait patiently.

**Comment by xinyuxiao** [comment]
thanks!

**Comment by xinyuxiao** [comment]
Hi @haiduo 

Sorry to ask you again, can you give me some information about the second step of multi_precision in '8,6,4,2'? Under below settings: --optimizer sgd --PSKD --lr_min 0 --repeated_aug --weight_decay 5e-5 --lr 5e-4 --pretrain, the top-1 of w1a1 is 61.88%, thank you very much!

**Comment by haiduo** [comment]
Sorry, I haven't experimented with including "w1a1" before, usually it's (w8a8, w6a6 ,w4a4, w2a2 or w4a4, w3a3, w2a2) but I guess adding "w1a1" will affect the overall performance.

**Comment by xinyuxiao** [comment]
Hi @haiduo

**Comment by xinyuxiao** [comment]
Hi @haiduo 

sorry, the edit is false, the top-1 acc of 61.88% is to w2a2, moreover the other quantized settings of w8a8 ,w4a4 and w2a2 are not high (76.65, 77.95, 77.80). can you give me some information about the second step of multi_precision in '8,6,4,2'?

**Comment by haiduo** [comment]
Hello @xinyuxiao，

I'm very busy recently, please wait a few days.

**Comment by haiduo** [comment]
Hi @xinyuxiao  ,

Not too busy lately - did you figure out that issue? Want to swap contact info?

**Comment by xinyuxiao** [comment]
Hi @haiduo 

Thank you very much. I want to know the training information details in the quantization stage on cifar10 of multi-precision and mixture precision.

**Comment by haiduo** [comment]
> Hi [@haiduo](https://github.com/haiduo)
> 
> Thank you very much. I want to know the training information details in the quantization stage on cifar10 of multi-precision and mixture precision.

Could you clarify if you are referring to the hyperparameter settings as the training details? For  hyperparameter settings, please refer to the following:

![Image](https://github.com/user-attachments/assets/0c81c64a-3ac1-482f-b6b3-f7d2eec1e5da)

**Comment by xinyuxiao** [comment]
@haiduo 
Thanks！I try it again. For ImageNet, is it similar to this configuration? Is the initial training optimizer still sgd? Is quantization using adam?

**Comment by haiduo** [comment]
> [@haiduo](https://github.com/haiduo) Thanks！I try it again. For ImageNet, is it similar to this configuration? Is the initial training optimizer still sgd? Is quantization using adam?

The ImageNet trainer uses Adam, as using SGD can lead to unstable training. The specific configuration is as follows：
ResNet-18, using 8-bit as the teacher, batch size = 256, Adam optimizer, CosineAnnealingLR scheduler, 90 epochs, double rounding, LSQ. Other hyperparameters are similar to cifar10.

**Comment by xinyuxiao** [comment]
ok, thanks!
